ExampleMod/Content/Buffs/ExampleFireDebuff.cs

```diff
@@ -0,0 +1,26 @@
+using ExampleMod.Common.Players;
+using ExampleMod.NPCs;
+using Terraria;
+using Terraria.ModLoader;
+
+namespace ExampleMod.Content.Buffs
+{
+	// Fire debuff is an example of a buff that causes constant loss of life.
+	// See ExamplePlayer.UpdateBadLifeRegen for more information.
+	public class ExampleFireDebuff : ModBuff
```

Comment:
I think it might be better if this was ExampleLifeRegenDebuff, as the vanilla fire debuffs just decrement HP while this debuff stops natural life regen.

Reply:
Changing the name again isn't a problem, but are you sure about the fire debuffs?
I checked the vanilla code and it is the same as in ExamplePlayer and the wiki says the fire debuff prevents life regeneration.

Reply:
Changing the name again isn't a problem, but are you sure about the fire debuffs? I checked the vanilla code and it is the same as in ExamplePlayer and the wiki says the fire debuff prevents life regeneration.

Reply:
> I think it might be better if this was ExampleLifeRegenDebuff, as the vanilla fire debuffs just decrement HP while this debuff stops natural life regen.

Changing the name again isn't a problem, but are you sure about the fire debuffs? I checked the vanilla code and it is the same as in ExamplePlayer and the wiki says the fire debuff prevents life regeneration.

Reply:
Oh I actually wasn't aware of that. But I feel that if a newer modder were to read "fire debuff", they would assume it deals damage, which this doesn't showcase. Of course, you could add that.